From b007d60ce6ea408f0b1bd0589d53455980072c13 Mon Sep 17 00:00:00 2001
From: Sandy Huang <hjc@rock-chips.com>
Date: Tue, 3 Dec 2019 21:08:36 +0800
Subject: [PATCH] drm/rockchip: drv: delete legacy fb setting

Primary plane for this CRTC, this is only relevant for legacy IOCTL,
it specifies the plane implicitly used by the SETCRTC and PAGE_FLIP IOCTLs,
It does not have any significance beyond that.

Change-Id: Ia4b0ac8fd4f597cd628377be32eef2ef498f4af3
Signed-off-by: Sandy Huang <hjc@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_drv.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_drv.c b/drivers/gpu/drm/rockchip/rockchip_drm_drv.c
index b2042b97da0a..2d776c31ed31 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_drv.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_drv.c
@@ -1563,10 +1563,8 @@ static int rockchip_drm_bind(struct device *dev)
 			break;
 
 		s = to_rockchip_crtc_state(crtc->state);
-		if (is_support_hotplug(s->output_type)) {
-			crtc->primary->fb = helper->fb;
+		if (is_support_hotplug(s->output_type))
 			drm_framebuffer_get(helper->fb);
-		}
 	}
 
 	drm_dev->mode_config.allow_fb_modifiers = true;
-- 
2.35.3

